File: backend-spring/src/main/java/com/nearkart/entity/DeliveryPlace.java
package com.nearkart.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "delivery_places")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Builder
public class DeliveryPlace {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(name = "company_id")
    private UUID companyId;

    @Column(name = "place_name", nullable = false)
    private String placeName;

    private String district;
    private String state;

    @Column(name = "created_at", nullable = false, updatable = false)
    private OffsetDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private OffsetDateTime updatedAt;

    @PrePersist
    protected void onCreate() { OffsetDateTime now = OffsetDateTime.now(); createdAt = now; updatedAt = now; }

    @PreUpdate
    protected void onUpdate() { updatedAt = OffsetDateTime.now(); }
}

File: backend/app/schemas/delivery_place.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class DeliveryPlaceCreate(BaseModel):
    company_id: UUID | None = None
    place_name: str
    district: str | None = None
    state: str | None = None


class DeliveryPlaceUpdate(BaseModel):
    company_id: UUID | None = None
    place_name: str | None = None
    district: str | None = None
    state: str | None = None


class DeliveryPlaceResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_id: UUID
    company_id: UUID | None
    place_name: str
    district: str | None
    state: str | None
    created_at: datetime
    updated_at: datetime
